### PR TITLE
Agenda - types fix

### PR DESCRIPTION
**Pull request by Inbal-Tish**
Agenda - fix types.
I added the Agenda demo screen to the type check and started fixing the types on all its related components.
Please double-check that I don't have any breaking changes!!

### CODE REVIEW

**Review by lidord-wix — COMMENTED** [review]
As far as I see you don't break anything since the changes are internal
I left two small comments

File: src/agenda/reservation-list/index.tsx [hunk 1]
```diff
@@ -136,15 +133,15 @@ class ReservationList extends Component<ReservationListProps, ReservationsListSt
...
-    });
+  updateDataSource(reservations: DayAgenda[]) {
+    this.setState({reservations});
   }
 
   updateReservations(props: ReservationListProps) {
     const {selectedDay} = props;
     const reservations = this.getReservations(props);
+    
+    if (selectedDay && this.selectedDay)
```
[the remark this answers is not in the document]
Comment (Inbal-Tish) [reply src/agenda/reservation-list/index.tsx:144]: I'm checking they are not undefined before passing them to 'sameDate()'

File: src/agenda/reservation-list/index.tsx [hunk 2]
```diff
@@ -182,14 +179,15 @@ class ReservationList extends Component<ReservationListProps, ReservationsListSt
...
       return {reservations: [], scrollPosition: 0};
     }
 
-    let reservations: DayReservations[] = [];
+    let reservations: DayAgenda[] = [];
     if (this.state.reservations && this.state.reservations.length) {
-      const iterator = this.state.reservations[0].day.clone();
-
+      const iterator = this.state.reservations[0].date?.clone();
+      if (iterator)
```
[the remark this answers is not in the document]
Comment (Inbal-Tish) [reply src/agenda/reservation-list/index.tsx:190]: The date is optional, so the iterator might be undefined. You need to check if it is not undefined before you use it.

File: src/agenda/reservation-list/index.tsx (hunk 1 above)
[the remark this answers is not in the document]
Comment (lidord-wix) [reply src/agenda/reservation-list/index.tsx:144]: I think this syntax is not very clear.. I suggest wrapping all the relevant scope with this condition :)

File: src/agenda/reservation-list/index.tsx (hunk 2 above)
[the remark this answers is not in the document]
Comment (lidord-wix) [reply src/agenda/reservation-list/index.tsx:190]: same here

[the remark this answers is not in the document]
Comment (Inbal-Tish) [reply src/agenda/reservation-list/index.tsx]: Done

File: src/agenda/reservation-list/index.tsx (hunk 1 above)
[the remark this answers is not in the document]
Comment (Inbal-Tish) [reply src/agenda/reservation-list/index.tsx]: Done

**Review by lidord-wix — APPROVED** [review]

**Pull request merged**